src/main/java/org/embl/mobie/io/ome/zarr/loaders/N5OMEZarrImageLoader.java

```diff
@@ -385,19 +414,42 @@ private VoxelDimensions getVoxelDimensions3D( int setupId, int datasetId ) {
             // get unit of last dimension, under assumption this is the X axis (i.e. a space axis)
             String unit = zarrAxesList.get(zarrAxesList.size() - 1).getUnit();
 
-            if ( scale != null && unit != null ) {
-                return new FinalVoxelDimensions(unit, scale);
+            if (scale != null && unit != null) {
+                if (zarrAxes.is2D() || zarrAxes.is3DWithChannels() ) {
+                    return new FinalVoxelDimensions(unit, new double[]{scale[0], scale[1], 1.0});
+                } else {
+                    return new FinalVoxelDimensions(unit, scale);
+                }
             }
         }
 
         return new DefaultVoxelDimensions(3);
     }
 
+    private FinalDimensions getFinalDimensions3D( int setupId ) {
+        final DatasetAttributes attributes = setupToAttributes.get(setupId);
+        if ( zarrAxes != null) {
+            if ( zarrAxes.hasChannels() ) {
+                long[] dims = attributes.getDimensions();
+                if ( zarrAxes.is3DWithChannels() ) {
+                    return new FinalDimensions(dims[0], dims[1], 1);
+                } else {
+                    return new FinalDimensions(dims[0], dims[1], dims[2]);
+                }
+            }
+        }
+
+        return new FinalDimensions(setupToAttributes.get(setupId).getDimensions());
+    }
+
     private ViewSetup createViewSetup(int setupId) {
         final DatasetAttributes attributes = setupToAttributes.get(setupId);
-        FinalDimensions dimensions = new FinalDimensions(attributes.getDimensions());
         OmeZarrMultiscales multiscale = setupToMultiscale.get(setupId);
+
+
+        FinalDimensions dimensions = getFinalDimensions3D(setupId);
```

Comment:
I am a bit confused by the code we had here before; it looks like beforehand we always just put the dimensions from the zarr directly, which might be 2 - 5D. Now we always put a 3D shape. The new code seems more reasonable to me, but it has worked in the old version, so I am not sure if this change is correct.
cc @KateMoreva